Index: OMOptimBasis/MOVector.cpp
===================================================================
--- OMOptimBasis/MOVector.cpp	(revision 17878)
+++ OMOptimBasis/MOVector.cpp	(working copy)
@@ -674,7 +674,7 @@
         if( !domItem.isNull() )
         {
             newItem = new ItemClass(domItem);
-            addItem(newItem);
+            this->MOAVector<ItemClass>::addItem(newItem);
         }
         n = n.nextSibling();
     }
@@ -705,7 +705,7 @@
     :MOAVector<ItemClass>(owner)
 {
     setItems(domList);
-    connect(this,SIGNAL(dataChanged(QModelIndex,QModelIndex)),this,SIGNAL(modified()));
+    this->QObject::connect(this,SIGNAL(dataChanged(QModelIndex,QModelIndex)),this,SIGNAL(modified()));
 }
 
 
@@ -738,7 +738,7 @@
         qDebug(QString("droped vector size :"+QString::number(dropedVector.size())).toLatin1().data());
         for(int i=0;i<dropedVector.size();i++)
         {
-            somethingDone = insertItem(dropedVector.at(i),row) || somethingDone;
+            somethingDone = this->MOAVector<ItemClass>::insertItem(dropedVector.at(i),row) || somethingDone;
         }
     }
     return somethingDone;

