diff --git a/OMNotebook/OMNotebookQT4/notebook.cpp b/OMNotebook/OMNotebookQT4/notebook.cpp
index af924a9..3bfc0ee 100644
--- a/OMNotebook/OMNotebookQT4/notebook.cpp
+++ b/OMNotebook/OMNotebookQT4/notebook.cpp
@@ -2627,7 +2627,7 @@ namespace IAEX
 	{
 		QString version = OmcInteractiveEnvironment::OMCVersion();
 		QString abouttext = QString("OMNotebook version 3.0 (for OpenModelica ") + version +
-			QString(")\r\n") + QString("Copyright 2004-2007, PELAB, Link" + QString(QChar(246, 0)) +"ping University\r\n\r\n") +
+			QString(")\r\n") + QString("Copyright 2004-2009, OSMC, Link" + QString(QChar(246, 0)) +"ping University\r\n\r\n") +
 			QString("Created by Ingemar Axelsson (2004-2005), Anders Fernstr" + QString(QChar(246, 0)) +"m (2005-2006) and Henrik Eriksson (2006-2007) as part of their final theses.");
 
 		QMessageBox::about( this, "OMNotebook", abouttext );
