--- Compiler/BackEnd/XMLDump.mo	(revision 11865)
+++ Compiler/BackEnd/XMLDump.mo	(working copy)
@@ -1917,7 +1917,7 @@
         dumpStrOpenTag(MathMLApply);
         dumpStrOpenTag(MathMLMatrix);
         dumpStrOpenTag(MathMLMatrixrow);
-        dumpListSeparator(ebs, dumpRow, stringAppendList({"\n</",MathMLMatrixrow,">/n<",MathMLMatrixrow,">"}));
+        dumpListSeparator(ebs, dumpRow, stringAppendList({"\n</",MathMLMatrixrow,">\n<",MathMLMatrixrow,">"}));
         dumpStrCloseTag(MathMLMatrixrow);
         dumpStrCloseTag(MathMLMatrix);
         dumpStrCloseTag(MathMLApply);
